pkg/apis/gcp/types_infrastructure.go

```diff
@@ -100,11 +104,28 @@ type CloudRouter struct {
 	Name string
 }
 
-// CloudNAT contains information about the the CloudNAT configuration
+// CloudNAT contains configuration about the the CloudNAT resource
 type CloudNAT struct {
 	// MinPortsPerVM is the minimum number of ports allocated to a VM in the NAT config.
 	// The default value is 2048 ports.
 	MinPortsPerVM *int32
+	// NatIPNames is a list of all names of user provided external premium ips which can be used by the nat gateway
+	// +optional
+	NatIPNames *[]NatIPName
```

Comment:
```suggestion
	NatIPNames []NatIPName
```

Reply:
Shouldn't optional paramters be pointer?